internal/migration/instance_service.go

```diff
@@ -38,26 +38,10 @@ func NewInstanceService(repo InstanceRepo, source SourceService, opts ...Service
 }
 
 func (s instanceService) Create(ctx context.Context, instance Instance) (Instance, error) {
-	var err error
+	// Note that we expect the source to fully populate an instance. For instance, the VMware source does
+	// this in its GetAllVMs() method.
```

Comment:
Do we keep the existing UUID from VMWare? If so, then this is fine. If there is no UUID, that we preserve, then I wonder, if it is the callers responsibility to provide a UUID or if this service should do it. Maybe we could add handling for the zero value UUID and generate one if none is provided.

In regards to the secret token, I don't think, this should be the callers responsibility. In this case, I think the service should generate it upon creation of a new instance.

Reply:
It's pretty important that we keep the same UUID coming from the source and onto the target.

Reply:
Yes, we're grabbing the UUID directly from VMware (https://github.com/FuturFusion/migration-manager/blob/main/internal/source/vmware.go#L126). With the current random UUID, entertainingly every time the source is scanned for updates duplicate instances are added to the database.

At the moment the VMware source is populating all the values for each returned instance (https://github.com/FuturFusion/migration-manager/blob/main/internal/source/vmware.go#L338). We could potentially move some common bits elsewhere, but it is nice to be guaranteed each instance returned is fully complete.